File: banka-frontend-enhanced/Banka-3-Backend-rewrite/pkg/account/sentinels.go
package account

// Sentinel UUIDs used as owner_client_id (bank) or client_id (trading)
// to identify bank-owned positions across the system. Kept in a shared
// package so both bank and trading services agree on the values without
// cross-importing each other's `internal/domain` package.
//
// Each ID maps 1:1 to a class of bank-managed accounts/positions:
//
//   - SystemOwnerID     — menjačnica house accounts (one per currency).
//   - StateTaxOwnerID   — the state's RSD capital-gains-tax account.
//   - ForexBookOwnerID  — bank's per-currency forex inventory book.
//   - BankAsClientOwnerID — client_id stamped on client_fund_positions
//     when the bank itself holds a stake in one of its own funds
//     (spec p.75 Napomena 2).
//   - FundsOwnerID      — owner_client_id stamped on every bank account
//     created to hold an investment fund's liquidity (spec p.74).
//
// Bank service's internal/domain re-exports the same values; the
// duplicate constants there exist for backwards-compat with code
// written before this shared package landed. New code references
// pkg/account.<Sentinel>.
const (
	SystemOwnerID       = "00000000-0000-0000-0000-000000000000"
	StateTaxOwnerID     = "00000000-0000-0000-0000-000000000010"
	ForexBookOwnerID    = "00000000-0000-0000-0000-000000000020"
	BankAsClientOwnerID = "00000000-0000-0000-0000-000000000030"
	FundsOwnerID        = "00000000-0000-0000-0000-000000000040"
)

File: banka-frontend-enhanced/src/lib/trading/sentinels.ts
// Sentinel UUIDs that the bank service uses as `owner_client_id` on
// bank-owned accounts. Mirrors the constants in
// `services/bank/internal/domain/domain.go`. Trading actuaries place
// orders against forex_book accounts (one per supported currency,
// pre-funded as the bank's open-FX position book), so the OrderForm
// queries on this sentinel instead of the principal's userId.

export const FOREX_BOOK_OWNER_ID = '00000000-0000-0000-0000-000000000020'

// BankAsClientOwnerID — `client_id` stamped on `client_fund_positions`
// when the bank itself holds a stake in one of its own funds (spec
// p.75 Napomena 2 — supervisor "investing in the name of the bank").
export const BANK_AS_CLIENT_OWNER_ID = '00000000-0000-0000-0000-000000000030'

// FundsOwnerID — `owner_client_id` on every bank account opened to
// hold an investment fund's RSD liquidity. Funds' own RSD accounts
// surface to supervisors via `listAccounts({ ownerClientId: this })`.
export const FUNDS_OWNER_ID = '00000000-0000-0000-0000-000000000040'
